### PR TITLE
ENG-10194; disable @StopNode while rejoining node still in process

### PR DESCRIPTION
**Pull request by JadenW**
add a blocker for each rejoin node (/db/rejoin_nodes_blocker/host_x) during rejoining process
I used hostId as the identifier for each blocker.. So I just used  CreateMode.EPHEMERAL.

### CODE REVIEW

File: src/frontend/org/voltdb/InvocationDispatcher.java [hunk 1]
```diff
@@ -650,12 +650,15 @@ private ClientResponseImpl dispatchStopNode(StoredProcedureInvocation task) {
         if (!m_cartographer.isClusterSafeIfNodeDies(liveHids, ihid)) {
             hostLog.info("Its unsafe to shutdown node with hostId: " + ihid
                     + " Cannot stop the requested node. Stopping individual nodes is only allowed on a K-safe cluster."
+                    + "And all rejoin nodes should be completed."
```
Comment (vtkstef) [inline src/frontend/org/voltdb/InvocationDispatcher.java]: add space between the double quote " and the capital A

File: src/frontend/org/voltdb/InvocationDispatcher.java [hunk 2]
```diff
@@ -650,12 +650,15 @@ private ClientResponseImpl dispatchStopNode(StoredProcedureInvocation task) {
...
                     + " Use shutdown to stop the cluster.");
             return gracefulFailureResponse(
                     "Cannot stop the requested node. Stopping individual nodes is only allowed on a K-safe cluster."
+                  + "And all rejoin nodes should be completed."
```
Comment (vtkstef) [inline src/frontend/org/voltdb/InvocationDispatcher.java]: same here

File: src/frontend/org/voltdb/VoltZK.java [hunk 3]
```diff
@@ -281,4 +285,36 @@ public static boolean removeCatalogUpdateBlocker(ZooKeeper zk, String node, Volt
         }
         return true;
     }
+
+    public static void createRejoinNodeIndicator(ZooKeeper zk, String node)
+    {
+        try {
+            System.out.println("start create znode" + node);
```
Comment (vtkstef) [inline src/frontend/org/voltdb/VoltZK.java]: remove

File: src/frontend/org/voltdb/VoltZK.java [hunk 4]
```diff
@@ -281,4 +285,36 @@ public static boolean removeCatalogUpdateBlocker(ZooKeeper zk, String node, Volt
...
+            zk.create(node,
+                      null,
+                      Ids.OPEN_ACL_UNSAFE,
+                      CreateMode.EPHEMERAL);
+            System.out.println("finshed create znode" + node);
```
Comment (vtkstef) [inline src/frontend/org/voltdb/VoltZK.java]: remove

File: src/frontend/org/voltdb/RealVoltDB.java [hunk 5]
```diff
@@ -2713,6 +2713,9 @@ public synchronized void recoveryComplete(String requestId) {
         if (m_rejoining) {
             if (m_rejoinTruncationReqId.compareTo(requestId) <= 0) {
                 String actionName = m_joining ? "join" : "rejoin";
+                // remove the rejoin blocker
```
Comment (vtkstef) [inline src/frontend/org/voltdb/RealVoltDB.java]: is recovery complete only called in the rejoining node?

File: src/frontend/org/voltdb/VoltZK.java [hunk 6]
```diff
@@ -281,4 +285,36 @@ public static boolean removeCatalogUpdateBlocker(ZooKeeper zk, String node, Volt
...
+            }
+        } catch (InterruptedException e) {
+            VoltDB.crashLocalVoltDB("Unable to create rejoin node Indicator", true, e);
+        }
+    }
+
+    public static boolean removeRejoinNodeIndicator(ZooKeeper zk, String node)
+    {
+        try {
+            ZKUtil.deleteRecursively(zk, node);
```
Comment (vtkstef) [inline src/frontend/org/voltdb/VoltZK.java]: why this one and the straight up delete?

File: src/frontend/org/voltdb/InvocationDispatcher.java (hunk 2 above)
Comment (JadenW) [inline src/frontend/org/voltdb/InvocationDispatcher.java]: fixed. thanks

File: src/frontend/org/voltdb/VoltZK.java (hunk 4 above)
Comment (JadenW) [inline src/frontend/org/voltdb/VoltZK.java]: removed

File: src/frontend/org/voltdb/RealVoltDB.java (hunk 5 above)
Comment (JadenW) [inline src/frontend/org/voltdb/RealVoltDB.java]: yes.

File: src/frontend/org/voltdb/VoltZK.java (hunk 6 above)
Comment (JadenW) [inline src/frontend/org/voltdb/VoltZK.java]: I will replace this to zk.delete.

**Pull request merged**